ntp-daemon/src/config/peer.rs

```diff
@@ -16,14 +16,52 @@ impl Default for PeerHostMode {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct PeerConfig {
+#[derive(Deserialize, Debug)]
+pub enum PeerType {
```

Comment:
I believe `PeerHostMode` was originally added as the discriminator for the same purpose, so I think we should remove that one. I would also add lowercase aliases for the variants in `PeerType` and/or `PeerHostMode`.